app/api/models.py

```diff
@@ -1,58 +1,72 @@
 """Data models."""
 
 from enum import Enum
-from typing import Optional, Union
+from typing import Annotated, Literal, Optional, Union
 
 from fastapi import Query
 from fastapi.exceptions import HTTPException
-from pydantic import BaseModel, constr, root_validator, validator
+from pydantic import BaseModel, BeforeValidator, model_validator
+from pydantic.types import StringConstraints
+from typing_extensions import Self
 
 CONTROLLED_TERM_REGEX = r"^[a-zA-Z]+[:]\S+$"
 VERSION_REGEX = r"^([A-Za-z0-9-]+)\.(\d+)\.([A-Za-z0-9-]+)$"
 
 
+def convert_valid_is_control_values_to_bool(
+    value: Optional[str],
+) -> Optional[bool]:
+    """
+    Convert case-insensitive values of 'true' to boolean True,
+    or raise a validation error for other string values.
+    """
+    if value is not None:
+        if value.lower() != "true":
+            raise HTTPException(
+                status_code=422,
+                detail="'is_control' must be either set to 'true' or omitted from the query",
+            )
+        return True
+    return None
+
+
 class QueryModel(BaseModel):
     """Data model and dependency for API that stores the query parameters to be accepted and validated."""
 
+    # TODO: If desired, we can explicitly forbid & error out on extra query parameters,
+    # following the syntax documented in https://fastapi.tiangolo.com/tutorial/query-param-models/#forbid-extra-query-parameters
+    # (at the moment, extra query parameters are just ignored/have no effect)
```

Comment:
Interesting. I think having extra parameters allowed is probably a good idea because otherwise we would have even stricter breaking changes. I.e. if we introduce a new query parameter, old APIs would start breaking immediately - probably don't want this.

So I think here it's a better idea to put the above comment into an issue with a discussion flag.

Reply:
I agree with you actually, so instead of an issue, maybe I'll just shorten this comment to explicitly describe the current behaviour when extra query params are provided.